agent/banner/banner.go

```diff
@@ -192,17 +192,23 @@ func (w *bannerResponseWriter) WriteHeader(statusCode int) {
 	if err != nil {
 		sc := http.StatusInternalServerError
 		http.Error(w, err.Error(), sc)
-		w.metricHandler.WriteResponseCodeMetric(sc)
+		if w.metricHandler != nil {
```

Comment:
We should not need any of these nil checks because the `WriteResponseCodeMetric` method correctly short-circuits if the receiver is nil.

Reply:
Done. Thanks for review!